src/inference/forwardSample.js

```diff
@@ -68,6 +70,12 @@ module.exports = function(env) {
     },
 
     factor: function(s, k, a, score) {
+      if (!this.opts.guide && !this.factorWarningIssued) {
```

Comment:
I think of the guide program as been the thing you get if you sample from guide distributions at `sample` statements and ignore `factor`. From this perspective, sampling from the guide doesn't include `factor`s by definition, and therefore `forward` isn't doing anything out of the ordinary that we need to warn people about.